src/System.Runtime.Numerics/src/System/Numerics/BigIntegerBuilder.cs

```diff
@@ -160,23 +160,11 @@ internal void GetIntegerParts(int signSrc, out int sign, out uint[] bits)
 
             int cuExtra = _rgu.Length - _iuLast - 1;
             Debug.Assert(cuExtra >= 0);
-            if (cuExtra <= 1)
+            if (cuExtra == 0)
             {
-                if (cuExtra == 0 || _rgu[_iuLast + 1] == 0)
```

Comment:
The `BigIntegerBuilder` implementation seems to allow one (and only one) leading zero. I tracked these cases down and found only a few occurrences: the `ctor` for `long` and `ulong`, which I've fixed. And two or three cases of `ModPow`, which I'm planning to replace with an "improved" version anyway.

This leading zero leads to cumbersome treating within `BigIntegerCalculator`, which was a bit confusing too (see older comments). Hence, getting rid of it seems to be the nicer solution (I added the according condition to `AssertValid`...). In short, it's nearly dead code and would become dead, if I'm continuing with this work.